network-config-analyzer/K8sServiceYamlParser.py

```diff
@@ -0,0 +1,157 @@
+#
+# Copyright 2020- IBM Inc. All rights reserved
+# SPDX-License-Identifier: Apache2.0
+#
+
+import Peer
+import yaml
+from PeerContainer import PeerContainer
+from K8sService import K8sService
+from CmdlineRunner import CmdlineRunner
+from GenericTreeScanner import TreeScannerFactory
+from K8sYamlParser import K8sYamlParser
+
+
+class K8sServiceYamlParser(K8sYamlParser):
+    """
+    A parser for k8s service resources
+    """
+
+    def __init__(self, service_file_name=''):
+        """
+        :param str service_file_name: The name of the yaml file containing K8s service resources
+        """
+        K8sYamlParser.__init__(self, service_file_name)
+
+    def parse_service(self, srv_object, peer_container):
+        """
+        Parses a service resource object and creates a K8sService object
+        :param dict srv_object: the service object to parse
+        :param PeerContainer peer_container: the peer container in which pods of the service are located
+        :return: K8sService object or None
+        """
+        if srv_object.get('kind') != 'Service':
+            return None  # Not a Service object
+        api_version = srv_object.get('apiVersion')
+        if api_version != 'v1':
+            return None  # apiVersion is not properly set
+        metadata = srv_object.get('metadata')
+        if not metadata:
+            return None
+        srv_name = metadata.get('name')
+        if not srv_name:
+            return None
+        self.check_dns_subdomain_name(srv_name, metadata)
```

Comment:
```suggestion
```
Can remove this check of valid resource name, as these checks are only applied to policy resources (similarly, pod resource name is not checked).

Reply:
Done